wordpress-mobile/WordPress-Android
# Fix the crash on reader post list screen

It crashes when you bookmark a post on a post list screen. 

 This crash was introduced with https://github.com/wordpress-mobile/WordPress-Android/pull/18158. `ReaderPostWebViewCachingFragment` was a Hilt fragment inside a non-Hilt activity. I migrated `ReaderPostListActivity` to Hilt to fix the crash.

Sentry URL: https://a8c.sentry.io/issues/4062245831/events/5edb39f3e0e649da88de8618ab40622a/

**Crash case:**

https://user-images.githubusercontent.com/2471769/230745203-bc9f7d9a-2925-4b68-a399-28d33494be3e.mp4

To test:
1. Launch the JP app.
2. Navigate to Reader.
3. Tap a blog's title to open a screen that lists its posts.
4. Tap the bookmark button on a post.
5. Ensure it doesn't crash.

## Regression Notes
1. Potential unintended areas of impact
None

6. What I did to test those areas of impact (or what existing automated tests I relied on)
Nothing

7. What automated tests I added (or what prevented me from doing so)
Added `e2eBookmarkPost` to test bookmarking on the post list screen.

PR submission checklist:
